File: ts/examples/hello-world.ts
// Minimal harness usage: invoke claude-code, print cost + result.
// Run: bun ts/examples/hello-world.ts  (from repo root)
// Or:  bun hello-world.ts              (from ts/examples/)
import { run } from '../src/index.js'
import { mkdtempSync } from 'node:fs'
import { join } from 'node:path'
import { tmpdir } from 'node:os'

const wd = mkdtempSync(join(tmpdir(), 'harness-'))

const r = await run({
  harness: 'claude-code',
  model: 'sonnet',
  prompt: "Print the string 'hello from harness' and nothing else.",
  workdir: wd,
  timeoutSeconds: 120,
})

const cost = r.costUsd != null ? `$${r.costUsd.toFixed(4)}` : 'n/a'
console.log(`exit=${r.exitCode}  cost=${cost}  tokens=${r.tokensIn}/${r.tokensOut}  wall=${r.durationSeconds.toFixed(1)}s`)
console.log(r.stdout.slice(0, 400))

File: examples/hello-world.py
#!/usr/bin/env python3
"""Minimal harness usage: invoke claude-code, print cost + result.

Run from repo root without installing:
    python examples/hello-world.py
"""
import os
import sys
import tempfile

# Make `harness` importable from a clean checkout (no pip install -e . required).
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', 'src'))

from harness import RunSpec, run

with tempfile.TemporaryDirectory() as wd:
    r = run(RunSpec(
        harness="claude-code",
        model="sonnet",
        prompt="Print the string 'hello from harness' and nothing else.",
        workdir=wd,
        timeout_seconds=120,
    ))

cost = f"${r.cost_usd:.4f}" if r.cost_usd is not None else "n/a"
print(f"exit={r.exit_code}  cost={cost}  tokens={r.tokens_in}/{r.tokens_out}  wall={r.duration_seconds:.1f}s")
print(r.stdout[:400])
